### PR TITLE
Add helper for checking Quaternions

### PR DESCRIPTION
**Pull request by jcarpent**
@jmirabel @wxmerkt Related to #1181, I've added some helpers to check whether the quaternions are normalized. In the Lie groups operations, I've set the default value to 1e-8.

I would like to go a bit further and let the final user choose these values by himself/herself thanks to global variables which would be registered inside Singleton. In this way, the user can change this value online, and it will only affect the Debug mode.
What do you think?

### CODE REVIEW

File: unittest/quaternion.cpp [hunk 1]
```diff
@@ -42,5 +42,24 @@ BOOST_AUTO_TEST_CASE(test_uniformRandom)
...
+  typedef Quaternion::Coefficients Vector4;
+  
+  for(int i = 0; i < 1e6; ++i)
+  {
+    Quaternion q;
+    q.coeffs() = Vector4::Random() + Vector4::Constant(2);
+    BOOST_CHECK(!quaternion::isNormalized(q));
+    
+    q.normalize();
+    BOOST_CHECK(quaternion::isNormalized(q));
```
Comment (wxmerkt) [inline unittest/quaternion.cpp:60]: Looks great - thank you for the quick fix. We could also add an explicit test against Vector4d::Zero()

**Comment by jmirabel** [comment]
> I would like to go a bit further and let the final user choose these values by himself/herself thanks to global variables which would be registered inside Singleton. In this way, the user can change this value online, and it will only affect the Debug mode.
> What do you think?

No sure it is necessary, but if you do it, you may also considered a preprocessor variable.

**Comment by jcarpent** [comment]
At which level do we need a preprocessor variable?

**Comment by jmirabel** [comment]
To set the default precision on the norm of unit norm vectors.

File: unittest/quaternion.cpp (hunk 1 above)
Comment (jcarpent) [reply unittest/quaternion.cpp:60]: Done
